### ISSUE TITLE
[BUG] bashtop with windows-wsl (debian buster)

### ISSUE THREAD

**Issue by 7k0l4n5k1** [aristocratos/bashtop#110]
hello aristocratos,

I like to use bashtop. it runs for me under wsl-windows (debian / buster) in a zsh (version 5.7.1) with byobu (5.112)/tmux (2.8). my plugin-manager is antigen (d1dd78b)
until version 0.8.23 it works fine. from 0.8.24 until 0.8.27 there are problems. after start of bashtop the individual frames will show. but they are empty. the windows are no longer filled with content. 

thanks a lot

tom

[bashtop.cfg.txt](https://github.com/aristocratos/bashtop/files/4599879/bashtop.cfg.txt)
[error.log](https://github.com/aristocratos/bashtop/files/4599880/error.log)
[locales.txt](https://github.com/aristocratos/bashtop/files/4599881/locales.txt)

![screenshot](https://user-images.githubusercontent.com/44008541/81420482-4ccaed00-9150-11ea-9d44-45b7aea6836e.jpg)

**Comment by 7k0l4n5k1** [comment]
after update to 0.8.28 same result. the frames are empty after starting.

**Comment by aristocratos** [comment]
In version 0.8.24 input reading was changed from using `read` to using `dd`.

Do you have all dependencies? Does `dd` behave normally on your system? Have you tried not running through `byobu` and `tmux`?

**Comment by 7k0l4n5k1** [comment]
i disabled byobu/tmux and worked with bash-shell instead of zsh.

Here are my dependencies
-> bash 5.0.3
-> dd is part of coreutils 8.30. (dd works)
-> grep is version 3.3
-> ps is from proccps 3.3.15
-> sed is version 4.7
-> awk is version 4.2.1
-> iostat is part of sysstat 12.0.3
-> curl is version 7.64


i download and install bashtop 0.8.29 with same result as before (empty frames)
0.8.23 is last version that works under wsl (debian buster). the issue is only under windows-wsl.

bashtop 0.8.29 works under native linux (raspbian) perfect!

**Comment by aristocratos** [comment]
Seems to be wsl not respecting the timeout set with `stty`. Tested with `read` instead of `dd` and got the same result, but works if setting the timeout with `read` instead of `stty`.

That's however not a good fix since I changed from `read` to `stty` and `dd` because `read` has a habit of stalling. And the timing calculations differ between the two methods, so won't work just replacing.

I would recommend making a bug report with microsoft that `stty` don't work as expected.

I will look for a workaround when I have time but it's unlikely I will find something that wouldn't negatively effect other systems.
